src/HotChocolate/Core/src/Types/Types/Attributes/SubscribeAndResolveAttribute.cs

```diff
@@ -34,6 +35,11 @@ public sealed class SubscribeAndResolveAttribute
                         ? typeRef.Type.ElementType
                         : null;
 
+                    if (rewritten is null && typeof(ISourceStream).IsAssignableFrom(typeRef.Type))
```

Comment:
this is code smelly :) I think the issue here is that we did not recognize this in our extended type.

Reply:
https://github.com/ChilliCream/hotchocolate/blob/21ea620c0d638a2b2a166098d667e8875c976363/src/HotChocolate/Core/src/Types/Internal/ExtendedType.Helper.cs#L158

Reply:
add source stream to the allowed list types